lib/qs/error_handler.rb

```diff
@@ -2,12 +2,20 @@ module Qs
 
   class ErrorHandler
 
-    attr_reader :exception, :daemon_data, :job
+    attr_reader :exception, :context
     attr_reader :error_procs
 
-    def initialize(exception, daemon_data, job = nil)
-      @exception, @daemon_data, @job = exception, daemon_data, job
-      @error_procs = @daemon_data.error_procs.reverse
+    def initialize(exception, context)
+      @exception = exception
+      daemon_data = context[:daemon_data]
+      @context = ErrorContext.new(
+        daemon_data,
+        context[:queue_redis_key],
+        context[:serialized_payload],
+        context[:job],
+        context[:handler_class]
+      )
```

Comment:
@jcredding not sure if you want to change this here or not, but it feels like the `ErrorContext` shouldn't be a struct.  It feels like you should just create it with a context hash.  Just a thought since you are manually building it from a hash anyway.